app/services/api/school_partnerships/create.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+module API::SchoolPartnerships
+  class Create
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :registration_year
+    attribute :school_ecf_id
+    attribute :lead_provider_ecf_id
+    attribute :delivery_partner_ecf_id
+
+    validates :registration_year, :school_ecf_id, :lead_provider_ecf_id, :delivery_partner_ecf_id, presence: true
+    validate :registration_period_exists
+    validate :lead_provider_exists
+    validate :school_exists
+    validate :school_is_not_cip_only
+    validate :school_is_eligible
+    validate :delivery_partner_exists
+    validate :lead_provider_delivery_partnership_exists
+    validate :school_partnership_does_not_already_exists
+    validate :school_training_period_exists
+
+    def create
+      return false unless valid?
+
+      SchoolPartnership.create!(
+        school:,
+        lead_provider_delivery_partnership:
+      ).tap { |school_partnership| accept_expressions_of_interest(school_partnership) }
+    end
+
+  private
+
+    def accept_expressions_of_interest(school_partnership)
+      expressions_of_interest = school_partnership.lead_provider_active_period.expressions_of_interest
+      expressions_of_interest.for_school(school).each do |expression_of_interest|
```

Comment:
is it worth checking those expressions don't have existing school partnerships for whatever reason 🤔

Reply:
Yeah good shout, I've updated - not sure about the `pending` terminology but it'll do for now